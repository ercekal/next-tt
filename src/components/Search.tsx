import React, {useState, useCallback} from 'react';
import { Input } from '@material-ui/core';
import { Dispatch } from 'redux';
import { useDispatch } from 'react-redux';
import { getResults } from '../store/actionCreators';


const Search = () => {
  const [searchText, setSearchText] = useState<string>('')
  const dispatch: Dispatch<any> = useDispatch();

  const handleSubmit = async (
    e: React.FormEvent<HTMLFormElement>
  ): Promise<void> => {
    e.preventDefault()
    dispatch(getResults(searchText))
  };
  return (
    <form onSubmit={handleSubmit}>
      <Input
        placeholder='Enter song, artist or album'
        onChange={(
          e: React.ChangeEvent<HTMLTextAreaElement>,
      ): void => setSearchText(e.target.value)}
        value={searchText}
      />
    </form>
  );
};

export default Search;