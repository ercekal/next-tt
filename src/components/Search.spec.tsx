import React from 'react';
import Search from './Search';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('Search component', () => {
  const mockStore = configureStore()
  const store = mockStore()
  it('renders correctly', () => {
  const tree = renderer
      .create(<Provider store={store}><Search /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

