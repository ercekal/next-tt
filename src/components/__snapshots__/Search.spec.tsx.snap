// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem component renders correctly 1`] = `
<div
  className="sc-bdnylx jkDdCw ResultItem"
>
  <p>
    Artist: 
    test artist
  </p>
  <p>
    Song Name: 
    test track
  </p>
  <p>
    Album: 
    test album
  </p>
</div>
`;

exports[`Search component renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Enter song, artist or album"
      type="text"
      value=""
    />
  </div>
</form>
`;
